queries.sql

```diff
@@ -29,3 +27,80 @@ SELECT * FROM animals WHERE name <> 'Gabumon';
 
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with weights that equal precisely 10.4kg or 17.3kg):
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+-- Verify the change (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that the species column went back to the original state (optional)
+SELECT * FROM animals;
+
+
+BEGIN;
+
+-- Update species to 'digimon' for animals whose name ends in 'mon'
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update species to 'pokemon' for animals without a species set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+-- Verify the changes (optional)
+SELECT * FROM animals;
+
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN;
+
+-- Delete all records in the animals table
+DELETE FROM animals;
+
+-- Verify that records were deleted (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that all records in the animals table still exist (optional)
+SELECT * FROM animals;
+
```

Comment:
- [ ] As suggested by the previous reviewer, kindly consider adding the following query that you missed:
  - Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction

Reply:
done